recipes/cpython/all/conanfile.py

```diff
@@ -512,6 +512,70 @@ def _msvc_package_copy(self):
         lib_dir_path = os.path.join(self.package_folder, self._msvc_install_subprefix, "Lib").replace("\\", "/")
         self.run(f"{interpreter_path} -c \"import compileall; compileall.compile_dir('{lib_dir_path}')\"")
 
+    def _exact_lib_name(self, folder):
+        possible_extensions = ("a", "so", "dylib", "lib")
+        for file in os.listdir(folder):
+            for extension in possible_extensions:
+                if re.match(f".*\\.{extension}", file):
+                    return file
```

Comment:
```suggestion
        patterns = ["*.a", "*.so*", "*.dylib", "*.lib"]
        for pattern in patterns:
            for file in Path(folder).glob(pattern):
                return file.name
```
Would be slightly nicer, imo. Needs a `from pathlib import Path` as well.

Reply:
Could maybe avoid the globbing altogether?
```py
def _exact_lib_name(self):
    if self.settings.os == "Windows":
        return self._lib_name if self.options.shared else f"{self._lib_name}.a"
    if is_apple_os(self):
        return f"lib{self._lib_name}.dylib" if self.options.shared else f"lib{self._lib_name}.a"
    return f"lib{self._lib_name}.so" if self.options.shared else f"lib{self._lib_name}.a"
```

Reply:
My original approach was to avoid the pattern matching, but I was concerned if there were any odd edge cases, for example using MinGW. Though I don't know if MinGW would even work for this recipe at all. Also I think on Windows it's a `.lib` regardless, but I'd have to double check.

Reply:
I do kind of like the non-globbed approach here, we can always fix if an issue comes up. I've mashed some of these implementations together, will push shortly.

Reply:
New implementation pushed